From b33fd247c76edae796a9e668de32dee66b798db8 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Mon, 14 May 2018 11:38:46 +0800
Subject: [PATCH 255/706] arm64: dts: ls1028a: Add Cadence display controller
 node

This patch adds Cadence display controller node for LS1028A.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit d8c1cd5d9e75a7ff98c3da3bf812866579ea76fd)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index f25c8981c10e..73439e6b5440 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -609,6 +609,48 @@
 			 <&aclk>, <&pclk>;
 		clock-names = "pxlclk", "mclk", "aclk", "pclk";
 		arm,malidp-output-port-lines = /bits/ 8 <8 8 8>;
+
+		port {
+			dp0_out: endpoint {
+				remote-endpoint = <&ip8501_out>;
+			};
+		};
+	};
+
+	ip8501: dp@f200000 {
+		compatible = "fsl,ls1028a-dp";
+		reg = <0x0 0xf200000 0x0 0x100000>;
+		interrupts = <0 221 0x4>;
+		clocks = <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&dpclk>,
+			 <&dpclk>,
+			 <&dpclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&pclk>,
+			 <&dpclk>,
+			 <&pclk>,
+			 <&pclk>;
+		clock-names = "dig_pll", "av_pll", "clk_ipg",
+			      "clk_core", "clk_pxl", "clk_pxl_mux",
+			      "clk_pxl_link", "clk_hdp", "clk_phy",
+			      "clk_apb", "clk_lis","clk_msi",
+			      "clk_lpcg", "clk_even","clk_dbl",
+			      "clk_vif", "clk_apb_csr","clk_apb_ctrl";
+
+		port {
+			ip8501_out: endpoint {
+				remote-endpoint = <&dp0_out>;
+			};
+		};
 	};
 };
 
-- 
2.17.1

